storm-core/src/clj/org/apache/storm/daemon/drpc.clj

```diff
@@ -210,67 +63,19 @@
-          ;; invocations that will unblock those threads
-          handler-server (when (> drpc-port 0)
-                           (ThriftServer. conf
-                             (DistributedRPC$Processor. drpc-service-handler)
-                             ThriftConnectionType/DRPC))
-          invoke-server (ThriftServer. conf
-                          (DistributedRPCInvocations$Processor. drpc-service-handler)
-                          ThriftConnectionType/DRPC_INVOCATIONS)
+          drpc-server (DrpcServer.)
           http-creds-handler (AuthUtils/GetDrpcHttpCredentialsPlugin conf)]
-      (Utils/addShutdownHookWithForceKillIn1Sec (fn []
-                                            (if handler-server (.stop handler-server))
-                                            (.stop invoke-server)))
-      (log-message "Starting Distributed RPC servers...")
-      (future (.serve invoke-server))
       (when (> drpc-http-port 0)
-        (let [app (-> (webapp drpc-service-handler http-creds-handler)
-                    requests-middleware)
-              filter-class (conf DRPC-HTTP-FILTER)
-              filter-params (conf DRPC-HTTP-FILTER-PARAMS)
-              filters-confs [(FilterConfiguration. filter-class filter-params)]
-              https-port (int (or (conf DRPC-HTTPS-PORT) 0))
-              https-ks-path (conf DRPC-HTTPS-KEYSTORE-PATH)
-              https-ks-password (conf DRPC-HTTPS-KEYSTORE-PASSWORD)
-              https-ks-type (conf DRPC-HTTPS-KEYSTORE-TYPE)
-              https-key-password (conf DRPC-HTTPS-KEY-PASSWORD)
-              https-ts-path (conf DRPC-HTTPS-TRUSTSTORE-PATH)
-              https-ts-password (conf DRPC-HTTPS-TRUSTSTORE-PASSWORD)
-              https-ts-type (conf DRPC-HTTPS-TRUSTSTORE-TYPE)
-              https-want-client-auth (conf DRPC-HTTPS-WANT-CLIENT-AUTH)
-              https-need-client-auth (conf DRPC-HTTPS-NEED-CLIENT-AUTH)]
-
-          (UIHelpers/stormRunJetty
-            (int drpc-http-port)
-            (reify IConfigurator (execute [this server]
-                                   (UIHelpers/configSsl server
-                                     https-port
-                                     https-ks-path
-                                     https-ks-password
-                                     https-ks-type
-                                     https-key-password
-                                     https-ts-path
-                                     https-ts-password
-                                     https-ts-type
-                                     https-need-client-auth
-                                     https-want-client-auth)
-                                   (UIHelpers/configFilter server (ring.util.servlet/servlet app) filters-confs))))))
-      (start-metrics-reporters conf)
-      (when handler-server
-        (.serve handler-server)))))
+        (let [app (-> (webapp drpc-server http-creds-handler)
```

Comment:
Can we please have a follow on JIRA to port the webapp portion over to java too?  Until it is done I don't consider DRPC moved to java.